packages/neuron-ui/src/components/History/index.tsx

```diff
@@ -46,51 +73,343 @@ const History = () => {
+        let status = item.status as string
+        if (status === 'success' && confirmations < CONFIRMATION_THRESHOLD) {
+          status = 'confirming'
+        }
+
+        return t(`history.${status}`)
+      },
+    },
+    {
+      title: t('history.table.operation'),
+      dataIndex: 'operation',
+      align: 'left',
+      render(_, idx, ___, ____, expandedRow) {
+        return <ArrowOpenRightIcon className={styles.arrow} data-is-open={expandedRow === idx} />
+      },
+    },
+  ]
+
+  const onActionBtnClick = useCallback(
+    (e: React.SyntheticEvent<HTMLButtonElement>) => {
+      const btn = (e.target as HTMLButtonElement)?.closest('button')
+      if (btn?.dataset?.hash && btn?.dataset?.action) {
+        switch (btn.dataset.action) {
+          case 'explorer': {
+            openExternal(`${getExplorerUrl(isMainnet)}/transaction/${btn.dataset.hash}`)
+            break
+          }
+          case 'detail': {
+            navigate(`${RoutePath.HistoryDetailPage}/${btn.dataset.hash}`)
+            break
+          }
+          default: {
+            // ignore
+          }
+        }
+      }
+    },
+    [isMainnet]
+  )
+
+  const rowExtendRender = (column: State.Transaction) => {
+    const { blockNumber, hash, description } = column
+    const confirmations = 1 + bestBlockNumber - +blockNumber
+    const confirmationsLabel = confirmations > 1000 ? '1,000+' : localNumberFormatter(confirmations)
+    const isSelected = localDescription.key === column.hash
+
     return (
-      <Stack className={styles.container}>
-        <div className={styles.tools}>
-          <SearchBox
-            value={keywords}
-            className={styles.searchBox}
-            styles={{
-              root: {
-                background: '#e3e3e3',
-                borderRadius: 0,
-                fontSize: '1rem',
-                border: '1px solid rgb(204, 204, 204)',
-                borderTopLeftRadius: 2,
-                borderBottomLeftRadius: 2,
-              },
-            }}
-            placeholder={t('history.search.placeholder')}
-            onChange={onKeywordsChange}
-            onSearch={onSearch}
-            iconProps={{ iconName: 'Search', styles: { root: { height: '18px' } } }}
-          />
-          <Button className={styles.searchBtn} type="default" label={t('history.search.button')} onClick={onSearch} />
-          <Button
-            className={styles.exportBtn}
-            type="primary"
-            disabled={isExporting}
-            onClick={onExport}
-            label={t('history.export-history')}
-          >
-            <Export />
-          </Button>
-        </div>
-        <div>
-          {totalCount ? (
-            <TransactionList
-              walletID={id}
-              walletName={walletName}
-              items={items as State.Transaction[]}
-              bestKnownBlockNumber={bestBlockNumber}
-              isMainnet={isMainnet}
-              dispatch={dispatch}
+      <tr>
```

Comment:
I think if you use `grid` layout, the element may be more clear.

Reply:
I had the same idea before, but in practice, it’s difficult to get the same width as each column in this list.